daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -192,6 +192,11 @@ trait ElementBaseGrammarMixin
         "%s is specified as a dfdl:prefixLengthType, but specifies a dfdl:trailingSkip other than 0",
         prefixLengthType
       )
+      schemaDefinitionWhen(
+        detachedElementDecl.statements.nonEmpty,
+        "%s is specified as a dfdl:prefixLengthType, but specifies one or more statement annotations",
```

Comment:
Can you update the message to include which "statement annotations" it contains. Users that aren't familiar with the term "statement annotation" (which sounds kindof generic so is understandable) might be confused what the problem is. If we instead say it contains "dfdl:assert", for example, that would be more useful.